wp/lib/bap_wp/src/environment.ml

```diff
@@ -64,7 +69,9 @@ and loop_handler = {
   handle : t -> Constr.t -> start:Graphs.Ir.Node.t -> Graphs.Ir.t -> t
 }
 
-and cond_type = Verify of Constr.goal | Assume of Constr.goal
+and cond = BeforeExec of Constr.goal | AfterExec of Constr.goal
+
+and cond_type = Verify of cond | Assume of cond
 
```

Comment:
I'm starting to worry that these are going to start nesting more and more as we come up with more attributes to apply to our conditions. No action needed now, but something to keep in mind before it gets out of hand.

Reply:
Sounds like a good idea